### PR TITLE
fix: Remove old querier

### PR DESCRIPTION
**Pull request by carols10cents**
Extracted from https://github.com/influxdata/influxdb_iox/pull/7049

This commit removes the old querier.

### CODE REVIEW

File: querier/src/table/mod.rs [hunk 1]
```diff
@@ -897,312 +822,6 @@ mod tests {
         assert_batches_eq!(&expected, &batches);
     }
 
-    #[tokio::test]
```
Comment (crepererum) [inline querier/src/table/mod.rs:900]: I would like to veto this and ask for more time to think about this and review it tomorrow. I'm not 100% convinced that we need no state reconciler and parquet file filtering at all anymore.

Also at least the last test (`test_parquet_cache_refresh`) is still relevant for the RPC mode.

File: querier/src/table/state_reconciler.rs [hunk 2]
```diff
@@ -118,19 +113,14 @@ impl Reconciler {
                 .push(tombstone);
         }
 
-        // Do not filter based on max sequence number in RPC write mode because sequence numbers
-        // are no longer relevant
```
Comment (crepererum) [inline querier/src/table/state_reconciler.rs:122]: Yeah I'm not sure this is actually correct.

File: querier/src/table/mod.rs (hunk 1 above)
Comment (crepererum) [reply querier/src/table/mod.rs:900]: Ok, so the cache invalidation test needs to be ported. The other two can be removed.

Comment (carols10cents) [reply querier/src/table/mod.rs:900]: Ok, I've ported the test and it found a bug in how this PR switched to only using the ingester Parquet file counts-- TL;DR I tried to collapse "no info from the ingesters" and "present but empty info from the ingesters" and they actually need to be treated differently. So there's a new commit for you to review here-- take all the time you need!

Comment (crepererum) [reply querier/src/table/mod.rs:900]: Just for my own understanding: does this mean we were invalidating caches in prod too often or not often enough?

**Review by crepererum — APPROVED** [review]
Thank you!

File: querier/src/table/mod.rs (hunk 1 above)
Comment (carols10cents) [reply querier/src/table/mod.rs:900]: I think prod is correct; it was my deletion in the first commit here (that has not been deployed yet) that was incorrect. The `else { false }` in the `remove_if` predicate is the important part.

**Pull request merged**